x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -37,21 +40,25 @@
  */
 public final class EnrichCache {
 
-    private final Cache<CacheKey, SearchResponse> cache;
+    private final Cache<CacheKey, CompletableFuture<SearchResponse>> cache;
     private volatile Metadata metadata;
 
     EnrichCache(long maxSize) {
-        this.cache = CacheBuilder.<CacheKey, SearchResponse>builder().setMaximumWeight(maxSize).build();
+        this.cache = CacheBuilder.<CacheKey, CompletableFuture<SearchResponse>>builder().setMaximumWeight(maxSize).build();
     }
 
-    SearchResponse get(SearchRequest searchRequest) {
+    CompletableFuture<SearchResponse> get(SearchRequest searchRequest) {
         String enrichIndex = getEnrichIndexKey(searchRequest);
         CacheKey cacheKey = new CacheKey(enrichIndex, searchRequest);
 
         return cache.get(cacheKey);
     }
 
     void put(SearchRequest searchRequest, SearchResponse searchResponse) {
+        put(searchRequest, CompletableFuture.completedFuture(searchResponse));
```

Comment:
These put methods can be removed and the `EnrichCacheTests` test needs to be converted to use `computeIfAbsent(...)` method.